examples/blas/compile_time_dispatching/level3/gemm_usm_mklgpu.cpp

```diff
@@ -0,0 +1,211 @@
+    // Execute Gemm
+    //
+
+    // add oneapi::mkl::blas::gemm to execution queue
+    try {
+        gemm_done = oneapi::mkl::blas::column_major::gemm(oneapi::mkl::backend_selector<oneapi::mkl::backend::mklgpu> {main_queue}, transA, transB, m, n, k, alpha, A, ldA, B, ldB, beta, C, ldC);
+    }
+    catch(sycl::exception const& e) {
+        std::cout << "\t\tCaught synchronous SYCL exception during GEMM:\n"
+                  << e.what() << std::endl << "OpenCL status: " << get_error_code(e) << std::endl;
+    }
+
+    gemm_done.wait();
+
+    //
+    // Post Processing
+    //
+
+    std::cout << "\n\t\tGEMM parameters:\n";
+    std::cout << "\t\t\ttransA = " << ( transA == oneapi::mkl::transpose::nontrans ? "nontrans" : ( transA == oneapi::mkl::transpose::trans ? "trans" : "conjtrans"))
+              <<   ", transB = " << ( transB == oneapi::mkl::transpose::nontrans ? "nontrans" : ( transB == oneapi::mkl::transpose::trans ? "trans" : "conjtrans")) << std::endl;
+    std::cout << "\t\t\tm = " << m << ", n = " << n << ", k = " << k << std::endl;
+    std::cout << "\t\t\tlda = " << ldA << ", ldB = " << ldB << ", ldC = " << ldC << std::endl;
+    std::cout << "\t\t\talpha = " << alpha << ", beta = " << beta << std::endl;
+
+
+    std::cout << "\n\t\tOutputting 2x2 block of A,B,C matrices:" << std::endl;
+
+    // output the top 2x2 block of A matrix
+    print_2x2_matrix_values(A, ldA, "A");
+
+    // output the top 2x2 block of B matrix
+    print_2x2_matrix_values(B, ldB, "B");
+
+    // output the top 2x2 block of C matrix
+    print_2x2_matrix_values(C, ldC, "C");
+
+    free(A, cxt);
+    free(B, cxt);
+    free(C, cxt);
+}
+
+//
+// Description of example setup, apis used and supported floating point type precisions
+//
+void print_example_banner() {
+
+    std::cout << "" << std::endl;
+    std::cout << "########################################################################" << std::endl;
+    std::cout << "# General Matrix-Matrix Multiplication using Unified Shared Memory Example: " << std::endl;
+    std::cout << "# " << std::endl;
+    std::cout << "# C = alpha * A * B + beta * C" << std::endl;
+    std::cout << "# " << std::endl;
+    std::cout << "# where A, B and C are general dense matrices and alpha, beta are" << std::endl;
+    std::cout << "# floating point type precision scalars." << std::endl;
+    std::cout << "# " << std::endl;
+    std::cout << "# Using apis:" << std::endl;
+    std::cout << "#   gemm" << std::endl;
+    std::cout << "# " << std::endl;
+    std::cout << "# Supported floating point type precisions:" << std::endl;
+    std::cout << "#   float" << std::endl;
+    std::cout << "# " << std::endl;
+    std::cout << "########################################################################" << std::endl;
+    std::cout << std::endl;
+
+}
+
+
+//
+// Main entry point for example.
+//
+int main (int argc, char ** argv) {
+    print_example_banner();
+
+    sycl::device dev = sycl::device(sycl::gpu_selector());
```

Comment:
`sycl::device dev ( sycl::gpu_selector())`

Reply:
Addressed in latest commits